File: Go/algorithm/CSES/1.sort_and_search/10.missing_coin_sum.go
package __sort_and_search

import "sort"

func missingCoinSum(coins []int) int {
	sort.Ints(coins)
	sum := 1
	for _, c := range coins {
		if c > sum {
			return sum
		}
		sum += c
	}

	return sum
}

File: C/CSES/2.Sorting and Searching/10.missing_coin_sum.cpp
//
// Created by ban on 15/1/25.
//

#include <algorithm>
#include <cstdio>

typedef long long ll;
constexpr int maxN = 2e5;

int main() {
    int n;

    scanf("%d", &n);

    ll coins[maxN] = {};
    for (int i = 0; i < n; i++) {
        scanf("%lld", &coins[i]);
    }

    std::sort(coins, coins + n);
    ll sum = 1;
    for (int i = 0; i < n; i++) {
        if (coins[i] > sum) {
            break;
        }
        sum += coins[i];
    }
    printf("%lld", sum);

    return 0;
}
